src/reanimated2/animations.js

```diff
@@ -1,37 +1,262 @@
+  switch (i % 6) {
+    case 0:
+      r = v, g = t, b = p;
+      break;
+    case 1:
+      r = q, g = v, b = p;
+      break;
+    case 2:
+      r = p, g = v, b = t;
+      break;
+    case 3:
+      r = p, g = q, b = v;
+      break;
+    case 4:
+      r = t, g = p, b = v;
+      break;
+    case 5:
+      r = v, g = p, b = q;
+      break;
+  }
+  return {
+    r: Math.round(r * 255),
+    g: Math.round(g * 255),
+    b: Math.round(b * 255)
+  };
+}
+
+export function transform(value, handler) {
+  'worklet';
+  if (value === undefined) {
+    return undefined;
+  }
+
+  if (typeof value === 'string') {
+    // toInt
+    // TODO handle color
+    const match = value.match(/(\D*)([\d.]*)(\D*)/);
+    const prefix = match[1];
+    const suffix = match[3];
+    const number = match[2];
+    handler.__prefix = prefix;
+    handler.__suffix = suffix;
+    return parseFloat(number);
+  }
+  
+  // toString if __prefix is available and number otherwise
+  if (handler.__prefix === undefined) {
+    return value;
+  }
+
+  return handler.__prefix + value + handler.__suffix;
+}
+
+const needsToBeChanged = ['toValue', 'current', 'startValue'];
+export function transformAnimation(animation) {
+  'worklet';
+  if (!animation) {
+    return;
+  }
+  needsToBeChanged.forEach((prop) => {
+    animation[prop] = transform(animation[prop], animation);
+  });
+}
+
+function convertToHSVA(color) {
+  'worklet';
+  const processedColor = processColorInitially(color); // argb;
+  const a = (processedColor >>> 24) / 255;
+  const r = (processedColor << 8) >>> 24;
+  const g = (processedColor << 16) >>> 24;
+  const b = (processedColor << 24) >>> 24;
+  const {
+    h,
+    s,
+    v
+  } = RGBtoHSV(r, g, b);
+  return [h, s, v, a];
+}
+
+function toRGBA(HSVA) {
+  'worklet';
+  const {
+    r,
+    g,
+    b
+  } = HSVtoRGB(HSVA[0], HSVA[1], HSVA[2]);
+  return `rgba(${r}, ${g}, ${b}, ${HSVA[3]})`;
+}
+
+export function decorateAnimation(animation) {
+  'worklet';
+  if (animation.isHigherOrder) {
+    return;
+  }
+  const baseOnStart = animation.onStart;
+  const baseOnFrame = animation.onFrame;
+  const animationCopy = Object.assign({}, animation); 
+  delete animation.callback;
+  
+  const prefNumberSuffOnStart = (animation, value, timestamp, previousAnimation) => {
+
+    const val = transform(value, animation);
+    transformAnimation(animation);
+    if (previousAnimation != animation) transformAnimation(previousAnimation);
+
+    baseOnStart(animation, val, timestamp, previousAnimation);
+
+    transformAnimation(animation);
+    if (previousAnimation != animation) transformAnimation(previousAnimation);
+  };
+  const prefNumberSuffOnFrame = (animation, timestamp) => {
+    transformAnimation(animation);
+
+    const res = baseOnFrame(animation, timestamp);
+
+    transformAnimation(animation);
+    return res;
+  };
+
+  const tab = ['H', 'S', 'V', 'A'];
+  const colorOnStart = (animation, value, timestamp, previousAnimation) => {
```

Comment:
would be nice to extract color-related helpers out of this file, it is already rather lengthy and adding so much stuff that is only relevant to some of the usecases does not seem to improve readability

Reply:
operation on colors have been moved to `Colors.js`. Here these two functions also contain operations on animations, I'd leave them here